src/main/java/com/epam/test_generator/services/TestResultService.java

```diff
@@ -0,0 +1,255 @@
+                    caseResult.setResult(Status.SKIPPED);
+                }
+                caseResult.setDuration(c.getDuration());
+                suitResult.setDuration(c.getDuration() + suitResult.getDuration());
+
+                final List<StepResult> collect = getStepResults(projectId, s, c);
+                caseResult.setSteps(collect);
+                caseResults.add(caseResult);
+            }
+
+            testResult.setDuration(testResult.getDuration() + suitResult.getDuration());
+
+            suitResult.setCases(caseResults);
+
+            final List<Status> collect = getStatusesFromSuitResult(suitResult);
+            suitResult.setStatus(evaluateStatusFrom(collect));
+
+            switch (suitResult.getStatus()) {
+                case PASSED:
+                    amountOfPassed++;
+                    break;
+                case SKIPPED:
+                    amountOfSkipped++;
+                    break;
+                case FAILED:
+                    amountOfFailed++;
+                    break;
+            }
+
+            suitResults.add(suitResult);
+        }
+
+        testResult.setSuits(suitResults);
+
+        final List<Status> collect = getStatusesFromEachSuitOfTestResult(testResult);
+        testResult.setStatus(evaluateStatusFrom(collect));
+
+        testResult.setAmountOfPassed(amountOfPassed);
+        testResult.setAmountOfFailed(amountOfFailed);
+        testResult.setAmountOfSkipped(amountOfSkipped);
+
+        return testResultDAO.save(testResult);
+    }
+
+    private List<Status> getStatusesFromSuitResult(SuitResult suitResult) {
+        return suitResult.getCases().stream().
+            map(CaseResult::getResult).
+            collect(Collectors.toList());
+    }
+
+    private List<StepResult> getStepResults(Long projectId, RawSuitResultDTO s,
+                                            RawCaseResultDTO c) {
+        return c.getSteps().stream()
+            .map(stepResultDTO -> getStepResult(projectId, s, c, stepResultDTO))
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Get all test results by projectId.
+     *
+     * @param projectId identifier of {@link Project}
+     * @return list of {@link TestResult}
+     */
+    public List<TestResultDTO> getTestResults(long projectId) {
+        return testResultDAO.findAllByProjectIdOrderByDateDesc(projectId).stream()
+            .map(testResultTransformer::toDto)
+            .collect(Collectors.toList());
+    }
+
+
+    /**
+     * Get limited amount of test results.
+     *
+     * @param projectId identifier of {@link Project}
+     * @param offset position of first element.
+     * @param limit position of last element.
+     * @return list of {@link TestResult}
+     */
+    public List<TestResultDTO> getTestResults(long projectId, int offset, int limit) {
+        return testResultDAO.findAllByProjectIdOrderByDateDesc(projectId).stream()
+            .skip(offset)
+            .limit(limit - offset)
```

Comment:
.limit(limit)

Limit is not upper boundary. Instead,  limit is number of objects required by client. 
If client wants to implement pagination, he shoul not care about upper boudary (ie 40-60 elements)
Instead client wants to start from 40 and get 20 more elements. It is important because if elements 50-55 was removed, querying 40-60 will return only 14 elements and that is not desired, client will have to make another query trying to guess correct value of upper boundary.